src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java

```diff
@@ -326,7 +330,7 @@ public final ResultsWrapper<ContentDTO> searchForContent(
             @Nullable final Set<String> topics, @Nullable final Set<String> books,
             @Nullable Set<String> levels, @Nullable Set<String> stages, @Nullable Set<String> difficulties,
             @Nullable Set<String> examBoards, final Set<String> contentTypes, final Integer startIndex,
-            final Integer limit, final boolean showNoFilterContent
+            final Integer limit, final boolean showNoFilterContent, final boolean fuzzySearch
```

Comment:
This is making me wonder if having global search and the finder/builder use the same method (`searchForContent`) is still the right thing to do, because it's getting quite cyclomatically complex based on the differing needs of each path (prior to [this commit](https://github.com/isaacphysics/isaac-api/commit/1e14c07341c3581417ed12c620bcf1247fd54a7d) they were separate). I can see them diverging further in future.

I think the value of `IsaacSearchInstructionBuilder` is being able to tell at a high level what we're searching for, and having a lot of dynamic stuff detracts from its' readability. What do you think @skyepurchase @mlt47 ?

Reply:
This sounds sensible. The plan would be to create a `siteWideSearch` function for `\search` endpoints and a `questionSearch` for `\questions` endpoints similar to the commit you shared. This should reduce the complexity of the functions and prevent unnecessary parameters being passed.

There's potentially a case for making the `SearchInstructionBuilder` more higher level, or having a higher level abstraction on top. But I don't that is necessary at the moment.